src/elliptic/curves/secp256_k1.rs

```diff
@@ -95,10 +95,20 @@ impl ECScalar for Secp256k1Scalar {
 
     fn new_random() -> Secp256k1Scalar {
         let mut arr = [0u8; 32];
-        thread_rng().fill(&mut arr[..]);
-        Secp256k1Scalar {
-            purpose: "random",
-            fe: SK::from_slice(&arr[0..arr.len()]).unwrap(),
+        let mut rng = thread_rng();
+
+        rng.fill(&mut arr[..]);
```

Comment:
I see here an opportunity to simplify the code:
```rust
loop {
    let mut arr = [0u8; 32];
    rng.fill(&mut arr[..]);
    match SK::from_slice(&arr[..]) {
        Ok(fe) => return Secp256k1Scalar {
            purpose: "random",
            fe,
        },
        Err(_) => continue,
    }
}
```

Reply:
Why not just use https://docs.rs/secp256k1/0.20.2/secp256k1/key/struct.SecretKey.html#method.new ?

Reply:
good point

Reply:
LOL
yep, better to use `new()`
will change it